tasks/agent.py

```diff
@@ -235,12 +235,13 @@ def omnibus_build(ctx, puppy=False, log_level="info", base_dir=None, gem_path=No
         ctx.run(cmd, env=env)
 
         omnibus = "bundle exec omnibus.bat" if invoke.platform.WINDOWS else "bundle exec omnibus"
-        cmd = "{omnibus} build {project_name} --log-level={log_level} {overrides}"
+        cmd = "{omnibus} build {project_name} --log-level={log_level} {populate_s3_cache} {overrides}"
         args = {
             "omnibus": omnibus,
             "project_name": "puppy" if puppy else "agent",
             "log_level": log_level,
-            "overrides": overrides_cmd
+            "overrides": overrides_cmd,
+            "populate_s3_cache": " --populate-s3-cache "
```

Comment:
Actually, using this flag makes the build fail when the build environment doesn't have permissions to push to S3. So let's make this a flag of the invoke command, and enable the flag in the gitlab build only.

Reply:
sounds like a good idea! I will fix this!